lib/collection.js

```diff
@@ -1874,6 +1893,59 @@ var count = function(self, query, options, callback) {
   });
 };
 
+/**
+ * Gets the number of documents matching the filter.
+ * @param {object} [query] the query for the count
+ * @param {object} [options] Optional settings.
+ * @param {object} [options.collation] Specifies a collation.
+ * @param {string|object} [options.hint] The index to use.
+ * @param {number} [options.limit] The maximum number of document to count.
+ * @param {number} [options.maxTimeMS] The maximum amount of time to allow the operation to run.
+ * @param {number} [options.skip] The number of documents to skip before counting.
+ * @param {Collection~countCallback} [callback] The command result callback.
+ * @return {Promise} returns Promise if no callback passed.
+ */
+
+Collection.prototype.countDocuments = function(query, options, callback) {
+  var args = Array.prototype.slice.call(arguments, 0);
+  callback = typeof args[args.length - 1] === 'function' ? args.pop() : undefined;
+  query = args.length ? args.shift() || {} : {};
+  options = args.length ? args.shift() || {} : {};
+
+  return executeOperation(this.s.topology, countDocuments, [this, query, options, callback]);
+};
+
+const countDocuments = function(self, query, options, callback) {
```

Comment:
`self` => `collection` (`self` describes very poorly outside of an attached function definition)